ibc/ibc_handle_event_test.go

```diff
@@ -0,0 +1,242 @@
+package ibc
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/pokt-network/pocket/persistence/kvstore"
+	"github.com/pokt-network/pocket/shared/messaging"
+	"github.com/pokt-network/pocket/shared/modules"
+	"github.com/stretchr/testify/require"
+)
+
+func TestHandleEvent_FlushCaches(t *testing.T) {
+	mgr, _, _, _, ibcMod := prepareEnvironment(t, 1, 0, 0, 0)
+	tmpDir := mgr.GetConfig().IBC.StoresDir + "/caches"
+	ibcHost := ibcMod.GetBus().GetIBCHost()
+	store, err := ibcHost.GetProvableStore("test")
+	require.NoError(t, err)
+
+	// set height
+	publishNewHeightEvent(t, ibcMod.GetBus(), 1)
+
+	kvs := []struct {
+		key   []byte
+		value []byte
+	}{
+		{
+			key:   []byte("key1"),
+			value: []byte("value1"),
+		},
+		{
+			key:   []byte("key2"),
+			value: []byte("value2"),
+		},
+		{
+			key:   []byte("key3"),
+			value: []byte("value3"),
+		},
+		{
+			key:   []byte("key4"),
+			value: nil,
+		},
+	}
+
+	for _, kv := range kvs {
+		if kv.value != nil {
+			require.NoError(t, store.Set(kv.key, kv.value))
+		} else {
+			require.NoError(t, store.Delete(kv.key))
+		}
+	}
+
+	// increment the height
+	publishNewHeightEvent(t, ibcMod.GetBus(), 2)
+
+	cache, err := kvstore.NewKVStore(tmpDir)
+	require.NoError(t, err)
+	keys, values, err := cache.GetAll([]byte{}, false)
+	require.NoError(t, err)
+	require.Len(t, keys, 4)
+	require.Len(t, values, 4)
+	for i, key := range keys {
+		require.Equal(t, string(key), prepareCacheKey(1, kvs[i].key))
+		if kvs[i].value == nil {
+			require.Equal(t, values[i], []byte{})
+			continue
+		}
+		require.Equal(t, values[i], kvs[i].value)
+	}
+
+	err = cache.ClearAll()
+	require.NoError(t, err)
+
+	newKeys, newValues, err := cache.GetAll([]byte{}, false)
+	require.NoError(t, err)
+	require.Len(t, newKeys, 0)
+	require.Len(t, newValues, 0)
+
+	require.NoError(t, cache.Stop())
+
+	// flush the cache
+	err = ibcHost.GetBus().GetBulkStoreCacher().FlushAllEntries()
+	require.NoError(t, err)
+
+	cache, err = kvstore.NewKVStore(tmpDir)
+	require.NoError(t, err)
+
+	// check in memory cache was cleared (ie nothing flushed)
+	newKeys, newValues, err = cache.GetAll([]byte{}, false)
+	require.NoError(t, err)
+	require.Len(t, newKeys, 0)
+	require.Len(t, newValues, 0)
+
+	require.NoError(t, cache.Stop())
+}
+
+// using MaxStoredHeight = 3
+func TestHandleEvent_PruneCaches(t *testing.T) {
+	mgr, _, _, _, ibcMod := prepareEnvironment(t, 1, 0, 0, 0)
+	tmpDir := mgr.GetConfig().IBC.StoresDir + "/caches"
+	ibcHost := ibcMod.GetBus().GetIBCHost()
+	store, err := ibcHost.GetProvableStore("test")
+	require.NoError(t, err)
+
+	kvs := []struct {
+		key   []byte
+		value []byte
+	}{
+		{
+			key:   []byte("key1"),
+			value: []byte("value1"),
+		},
+		{
+			key:   []byte("key2"),
+			value: []byte("value2"),
+		},
+		{
+			key:   []byte("key3"),
+			value: []byte("value3"),
+		},
+		{
+			key:   []byte("key4"),
+			value: nil,
+		},
+	}
+
+	testCases := []struct {
+		name            string
+		heights         []uint64
```

Comment:
Can you make this more explitict or add a comment?

1. `heightsStored` or `heights [] uint64 // the heights that have been committed to the state`